### PR TITLE
Add dockerfiles for deploy servers

### PR DESCRIPTION
**Pull request by asalkeld**

### CODE REVIEW

File: cloud/azure/deploy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+FROM python:3.9-slim AS base
+
+LABEL "repository"="https://github.com/nitrictech/nitric"
+LABEL "homepage"="https://nitric.io"
+LABEL "maintainer"="Nitric Team <team@nitric.io>"
+LABEL org.opencontainers.image.description="Nitric azure deployer, in a Docker container."
+
+ADD runtime-azure /usr/bin/
```
Comment (HomelessDinosaur) [inline cloud/azure/deploy/Dockerfile:8]:
```suggestion
ADD deploy-azure /usr/bin/
```

File: cloud/azure/deploy/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# Passing --build-arg PULUMI_VERSION=vX.Y.Z will use that version
+# of the SDK. Otherwise, we use whatever get.pulumi.com thinks is
+# the latest
+ARG PULUMI_VERSION
+
+# Install the Pulumi SDK, including the CLI and language runtimes.
+RUN curl -fsSL https://get.pulumi.com/ | bash -s -- --version $PULUMI_VERSION && \
+  mv ~/.pulumi/bin/* /usr/bin
+
+ENTRYPOINT ["runtime-azure"]
```
Comment (HomelessDinosaur) [inline cloud/azure/deploy/Dockerfile:46]:
```suggestion
ENTRYPOINT ["deploy-azure"]
```

File: Makefile [hunk 3]
```diff
@@ -9,6 +9,10 @@ binaries: $(providers)
 		$(MAKE) -C $$dir || exit 1; \
 	done
 
+dockerimage:
+	$(MAKE) dockerimage -C cloud/aws
+	$(MAKE) dockerimage -C cloud/azure
```
Comment (HomelessDinosaur) [inline Makefile:14]:
```suggestion
	$(MAKE) dockerimage -C cloud/azure
	$(MAKE) dockerimage -C cloud/gcp
```

**Review by HomelessDinosaur — CHANGES_REQUESTED** [review]

File: cloud/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  wget \
+  unzip && \
+  # Get all of the signatures we need all at once.
+  curl -fsSL https://download.docker.com/linux/debian/gpg          | apt-key add - && \
+  # Add additional apt repos all at once
+  echo "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable"      | tee /etc/apt/sources.list.d/docker.list           && \
+  # Install second wave of dependencies
+  apt-get update -y && \
+  apt-get install -y \
+  docker-ce && \
```
Comment (tjholm) [inline cloud/Dockerfile:27]: Thinking of slimming things down a bit. Do we actually need full docker-ce? Or could we get away with just docker-cli with the socket forwarding we'll need to do anyway?

File: cloud/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,38 @@
+FROM python:3.9-slim AS base
```
Comment (tjholm) [inline cloud/Dockerfile:1]: Do we need python?

here is pulumi's base dockerfile sans SDKs:
https://github.com/pulumi/pulumi-docker-containers/blob/main/docker/base/Dockerfile

I don't think we need the SDKs when we're using the automation API. Excluding unnecessary runtimes should save some space

File: cloud/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  apt-get install -y \
+  apt-transport-https \
+  build-essential \
+  ca-certificates \
+  curl \
+  git \
+  gnupg \
+  software-properties-common \
+  wget \
+  unzip && \
```
Comment (tjholm) [inline cloud/Dockerfile:19]: If we use pulumi/pulumi-base we should have everything we need that they rely on.

**Comment by tjholm** [comment]
Going with binary distribution approach for now.

**Pull request closed without merging**